### DISCUSSION TITLE
Switch to change the current program to silent mode (Bosch Dishwasher)

### DISCUSSION THREAD

**Discussion by Bingo269** [thoukydides/homebridge-homeconnect#154] [Ideas]
My new Bosch dishwasher has a feature to turn the currently running program into a silent mode so that e.g. if you are watching tv the dishwasher sounds does not distract you.
The feature can be turned on through the Home Connect app at any time.

It would be great to have a switch exposed to homekit to have that feature as well triggered by automations.
Currently, it is only possible to start new dedicated silent programs but not to turn a running program silent.

Cheers

**Comment by thoukydides** [comment]
That will be the [Dishcare.Dishwasher.Option.SilenceOnDemand](https://api-docs.home-connect.com/programs-and-options/?exclude=cleaningRobot%2CcoffeeMachine%2CcookProcessor%2Ccooktop%2Cdryer%2Cfreezer%2CfridgeFreezer%2Chood%2Coven%2Crefrigerator%2CwarmingDrawer%2Cwasher%2CwasherDryer%2CwineCooler#dishwasher_silence-on-demand-option) program option.

According to the [API documentation](https://apiclient.home-connect.com/#/programs/set_active_program_option) it is not possible to change dishwasher program options after the program has been started:
> Set one specific option of the active program
> This operation can be used to modify one specific option of the active
program, e.g. to extend the duration of the active program by
another 5 minutes.
> **Please note that changing options of the running program is currently only
supported by ovens.**

[and](https://apiclient.home-connect.com/#/programs/set_active_program_options):
> Set all options of the active program, e.g. to switch from preheating to the actual program options.
> Update the options for the currently running program.
With this API endpoint, you have to provide all options with
their new values. If you want to update only one option, you can use the
endpoint specific to that option.
> **Please note that changing options of the running program is currently only
supported by ovens.**

That matches my previous experience.

I am not aware of any API method to determine which options on which appliances can be changed while a program is active… So even if this is supported by the API (contrary to the documentation) then implementing this would probably need to be a manual configuration; similar to the way that program switches are currently configured. However, it wouldn’t be possible to create a configuration schema for the homebridge-config-ui-x GUI that limited it to options that would actually work… and generalising for things like oven program options would be complicated, since the available options and ranges would be program-specific, whereas the switch would apply to all programs.

**Comment by thoukydides** [comment]
Testing with my own Siemens dishwasher…

The `Dishcare.Dishwasher.Option.SilenceOnDemand` option is not supported. Attempting to set it returns the error `Program option not supported`. Attempting to set any supported option whilst a program is running returns the error `XXX currently not available or writable`.

It is possible that other appliances behave differently, but I would have no way to test and debug that functionality. Hence, I don’t currently plan to pursue this feature.
